### PR TITLE
fix(kuma-cp) resync IPAM in case of Kuma CP restart

### PR DESCRIPTION
**Pull request by lobkovilya**
Signed-off-by: Ilya Lobkov <ilya.lobkov@konghq.com>

### Summary

IPAM has to be synced between Kuma CP restarts. This PR fixes it

### CODE REVIEW

File: pkg/dns/ip.go [hunk 1]
```diff
@@ -2,13 +2,19 @@ package dns
...
 )
 
 type IPAM interface {
 	AllocateIP() (string, error)
 	FreeIP(ip string) error
+	ReserveIP(ip string) error
+}
+
+func IsAddressAlreadyAllocated(err error) bool {
+	return err != nil && strings.HasPrefix(err.Error(), "Address already allocated")
```
Comment (nickolaev) [inline pkg/dns/ip.go:17]: Hmm, can we have a custom ErrorType and compare against it here? Checking string prefix is probably not the best.

File: pkg/plugins/runtime/k8s/controllers/configmap_controller.go [hunk 2]
```diff
@@ -41,6 +41,10 @@ func (r *ConfigMapReconciler) Reconcile(req kube_ctrl.Request) (kube_ctrl.Result
 		return kube_ctrl.Result{}, nil
 	}
 
+	if err := r.VIPsAllocator.Sync(); err != nil {
```
Comment (nickolaev) [inline pkg/plugins/runtime/k8s/controllers/configmap_controller.go:44]: Hmm, why is this one here?

File: pkg/dns/ip.go (hunk 1 above)
Comment (lobkovilya) [reply pkg/dns/ip.go:17]: We already have the same approach in `store.go` and here error comes from `Nordix/simple-ipam` library. The only thing we can do is to check it against string literal.

File: pkg/plugins/runtime/k8s/controllers/configmap_controller.go (hunk 2 above)
Comment (lobkovilya) [reply pkg/plugins/runtime/k8s/controllers/configmap_controller.go:44]: `VIPsAllocator#Start` is called only on Universal. On Kubernetes we call `VIPsAllocator#CreateOrUpdateVIPConfig`by reacting on changes in the cluster.

**Review by jakubdyszkiewicz — APPROVED** [review]

**Pull request merged**